docs/dcp2_system_design.rst

```diff
@@ -986,6 +1065,400 @@ Broad Institute).
+by undoing the steps taken to add the alteration. For example, if a delta
+staging area adds a cell suspension to a project, reverting the addition prior
+to an import simply involves deleting the cell suspension entity object and
+any subgraphs that reference it. Reverting the addition after the staging area
+was imported means creating a new delta staging area with a removal marker
+object for the cell suspension entity and updated subgraph objects that omit
+any reference to the cell suspension.
+
+For alterations of metadata entities, subgraphs, descriptors and data files,
+we define two copies: the altered copy and the original. The
+*altered* copy is the one contained in the (delta) staging area about to be
+imported. The *original* is the copy that was most recently imported into TDR,
+the one that's currently present in the TDR dataset. Under this definition, a
+deletion or removal marker object is considerd an altered copy. It is
+insignificant if the original was imported from a staging area at the same URI
+as the one containing the altered copy, or a different URI.
+
+We further differentiate between removals and deletions. Data files can be
+deleted or removed, subgraphs and metadata entities can only be removed. A
+removed data file, subgraph or metadata entity disappears from subsequently
+created snapshots but remains present in previously created snapshots. A
+deleted data file is both removed from subsequently created snapshots **and**
+erased physically from storage, references to it in previously created
+snapshots become invalid and requesting the data file's content results in an
+error.
+
+
+Update an entity
+~~~~~~~~~~~~~~~~
+
+Write the updated entity document to an object in the ``metadata`` directory
+of the staging area. The name of that object must have the same
+``entity_type`` and ``entity_id`` as the original and a strictly higher
+``version``.
+
+
+Rename or update a data file
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Write the updated data file to an object in the ``data`` directory of the
+   staging area, using the same or a different file name as the original data
+   file.
+
+2. Write the updated descriptor to an object in the ``descriptors`` directory
+   of the staging area. The name of the descriptor object must have the same
+   ``entity_type`` and ``entity_id`` as the original, and a strictly higher
+   ``version``. The ``file_id`` property of the descriptor document must be
+   the same as that of the original, and the ``file_version`` property must be
+   strictly higher. The ``file_name`` property can have the same value or a
+   different one. Note that the ``file_id``property of the descriptor document
+   is independent from the column of the same name in TDR ``…_file`` tables and
+   that this section only applies to the former.
+
+   If the ``file_name`` property is the same as in the original and the
+   staging area has the `is_delta`` property set to ``true`` (``false``), the
```

Comment:
Does this apply if the `is_delta` property is set to `false`? The wording here is a bit confusing

Reply:
I think I see what is meant here, but still find the style confusing. Consider making the false case a separate sentence for clarity

Reply:
I love this style because it so concise. The next sentence clarifies the intent. I'll adopt the suggestion, if others also find the style confusing.

Reply:
Addressed. Added "or" in front.